### PR TITLE
Convert RandomPlant to entity system

### PR DESCRIPTION
**Pull request by wrexbe**
## About the PR
Converted the random plant component into ECS

### CODE REVIEW

File: Content.Server/Plants/Systems/RandomPottedPlantSystem.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            "plant-21", "plant-22", "plant-23", "plant-24", "applebush"
+        };
+        private static readonly string[] PlasticPlantStates =
+        {
+            "plant-26", "plant-27", "plant-28", "plant-29"
+        };
+
+        public override void Initialize()
+        {
+            SubscribeLocalEvent<RandomPottedPlantComponent,MapInitEvent>(OnMapInit);
```
Comment (ShadowCommander) [inline Content.Server/Plants/Systems/RandomPottedPlantSystem.cs:28]:
```suggestion
            SubscribeLocalEvent<RandomPottedPlantComponent, MapInitEvent>(OnMapInit);
```

File: Content.Server/Plants/Components/RandomPottedPlantComponent.cs [hunk 2]
```diff
@@ -1,80 +1,23 @@
...
-
-        protected override void Startup()
-        {
-            base.Startup();
-
-            if (_selectedState != null)
-            {
-                _entMan.GetComponent<SpriteComponent>(Owner).LayerSetState(0, _selectedState);
-            }
-        }
```
Comment (ShadowCommander) [inline Content.Server/Plants/Components/RandomPottedPlantComponent.cs:68]: Does it show the right state with a state defined in YAML without this?

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Content.Server/Plants/Components/RandomPottedPlantComponent.cs [hunk 3]
```diff
@@ -1,80 +1,23 @@
...
 namespace Content.Server.Plants.Components
 {
     [RegisterComponent]
-    public class RandomPottedPlantComponent : Component, IMapInit
+    [Friend(typeof(RandomPottedPlantSystem))]
+    public class RandomPottedPlantComponent : Component
     {
-        [Dependency] private readonly IEntityManager _entMan = default!;
-
         public override string Name => "RandomPottedPlant";
```
Comment (metalgearsloth) [inline Content.Server/Plants/Components/RandomPottedPlantComponent.cs:13]: There's also the funny new component attribute with the name.

File: Content.Server/Plants/Components/RandomPottedPlantComponent.cs (hunk 2 above)
Comment (PaulRitter) [reply Content.Server/Plants/Components/RandomPottedPlantComponent.cs:68]: yes, due to the ??= operator in mapinit

**Review by PaulRitter — APPROVED** [review]
lgtm besides the other review comments

**Review by ShadowCommander — APPROVED** [review]

**Pull request merged**